build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/BuildView.java

```diff
@@ -107,6 +115,21 @@ public String description() {
         return getOrElse(build.getDescription(), "");
     }
 
+    @Override
+    public List<String> stages() {
+        if (this.build instanceof WorkflowRun) {
+            WorkflowRun currentBuild = (WorkflowRun) this.build;
+
+            if (currentBuild.isBuilding()) {
+                StageTraversal traversal = new StageTraversal();
+                traversal.start(currentBuild.getExecution().getCurrentHeads());
```

Comment:
I tried this locally, and this row causes an NPE occasionally.

I think it is when the build monitor displays non-running jobs. In that case, I think getExecution() returns null.